flink-staging/flink-ml/src/main/scala/org/apache/flink/ml/classification/MultinomialNaiveBayes.scala

```diff
@@ -0,0 +1,900 @@
+
+  import MultinomialNaiveBayes._
+
+  //The model, that stores all needed information that are related to one specific word
+  var wordRelatedModelData: Option[DataSet[(String, String, Double)]] =
+    None // (class name -> word -> log P(w|c))
+
+  //The model, that stores all needed information that are related to one specifc class+
+  var classRelatedModelData: Option[DataSet[(String, Double, Double)]] =
+    None // (class name -> p(c) -> log p(w|c) not in class)
+
+  //A data set that stores additional needed information for some of the improvements
+  var improvementData: Option[DataSet[(String, Double)]] =
+    None // (word -> log number of documents in all classes / word frequency in all classes
+
+  // ============================== Parameter configuration ========================================
+
+  def setP1(value: Int): MultinomialNaiveBayes = {
+    parameters.add(P1, value)
+    this
+  }
+
+  def setP2(value: Int): MultinomialNaiveBayes = {
+    parameters.add(P2, value)
+    this
+  }
+
+  def setP3(value: Int): MultinomialNaiveBayes = {
+    parameters.add(P3, value)
+    this
+  }
+
+  def setSR1(value: Int): MultinomialNaiveBayes = {
+    parameters.add(SR1, value)
+    this
+  }
+
+  def setS1(value: Int): MultinomialNaiveBayes = {
+    parameters.add(S1, value)
+    this
+  }
+
+  def setR1(value: Int): MultinomialNaiveBayes = {
+    parameters.add(R1, value)
+    this
+  }
+
+  // =============================================== Methods =======================================
+
+  /**
+   * Save already existing model data created by the NaiveBayes algorithm. Requires the designated
+   * locations. The saved data is a representation of the [[wordRelatedModelData]] and
+   * [[classRelatedModelData]].
+   * @param wordRelated, the save location for the wordRelated data
+   * @param classRelated, the save location for the classRelated data
+   */
+  def saveModelDataSet(wordRelated: String, classRelated: String) : Unit = {
+    wordRelatedModelData.get.writeAsCsv(wordRelated, "\n", "|", WriteMode.OVERWRITE)
+    classRelatedModelData.get.writeAsCsv(classRelated, "\n", "|", WriteMode.OVERWRITE)
+  }
+
+  /**
+   * Save the improvment data set. Requires the designated save location. The saved data is a
+   * representation of the [[improvementData]] data set.
+   * @param path, the save location for the improvment data
+   */
+  def saveImprovementDataSet(path: String) : Unit = {
+    improvementData.get.writeAsCsv(path, "\n", "|", WriteMode.OVERWRITE)
+  }
+
+  /**
+   * Sets the [[wordRelatedModelData]] and the [[classRelatedModelData]] to the given data sets.
+   * @param wordRelated, the data set representing the wordRelated model
+   * @param classRelated, the data set representing the classRelated model
+   */
+  def setModelDataSet(wordRelated : DataSet[(String, String, Double)],
+                      classRelated: DataSet[(String, Double, Double)]) : Unit = {
```

Comment:
This should be written like
```scala
def setModelDataSet(
      wordRelated : DataSet[(String, String, Double)],
      classRelated: DataSet[(String, Double, Double)])
    : Unit = {
    this.wordRelatedModelData = Some(wordRelated)
    this.classRelatedModelData = Some(classRelated)
  }
```
Have a look at the existing code for indentation conventions.